### PR TITLE
Fix/ethers v6 migration in hardhat

### PR DESCRIPTION
**Pull request by swellander**
## Description
Updates `ethers` from `v5.7.1` => `v6.10.0`, mirroring the recent changes made to SE-2 in [PR #692](https://github.com/scaffold-eth/scaffold-eth-2/pull/692)

Your ENS/address: `websurfer.eth`

### CODE REVIEW

File: packages/hardhat/deploy/00_deploy_your_contract.ts [hunk 1]
```diff
@@ -31,7 +32,7 @@ const deployYourContract: DeployFunction = async function (hre: HardhatRuntimeEn
     autoMine: true,
   });
 
-  // const yourContract = await hre.ethers.getContract("YourContract", deployer);
+  // await hre.ethers.getContract<Contract>("YourContract", deployer);
```
Comment (carletex) [inline packages/hardhat/deploy/00_deploy_your_contract.ts:35]: Maybe change `YourContract` with `BatchRegistry`? So the example works out of the box when uncommenting.

**Comment by carletex** [comment]
Left a tiny comment. But looking great, thanks!!

File: packages/hardhat/deploy/00_deploy_your_contract.ts (hunk 1 above)
Comment (swellander) [reply packages/hardhat/deploy/00_deploy_your_contract.ts:35]: Ah yeah, good call :+1:

**Comment by swellander** [comment]
> Left a tiny comment. But looking great, thanks!!

@carletex thanks for the review! Just addressed your comment.

**Comment by carletex** [comment]
Amazing!! Thank you

**Pull request merged**